src/main.rs

```diff
@@ -107,7 +146,8 @@ async fn setup_hive(starknet_provider: &JsonRpcClient<HttpTransport>) -> Result<
     use starknet::{accounts::ConnectedAccount, core::types::Felt, providers::Provider as _};
 
     let chain_id = starknet_provider.chain_id().await?;
-    let chain_id: u64 = (Felt::from(u64::MAX).to_bigint() & chain_id.to_bigint()).try_into()?;
+    let modulo = 1u64 << 53;
+    let chain_id: u64 = (Felt::from(modulo).to_bigint() & chain_id.to_bigint()).try_into()?;
```

Comment:
Same here maybe a modulo operation is enough but I think this is applicable to the whole codebase so that maybe a follow up issue is appropriate

Reply:
https://github.com/kkrt-labs/kakarot-rpc/issues/1408 done!